### PR TITLE
First working version of osqp-matlab-cmake-buildsystem

### PR DESCRIPTION
**Pull request by traversaro**
Most of the relevant information is provided in the README. This is convenient to easily add the osqp-matlab to the robotology-superbuild, to fix https://github.com/robotology/robotology-superbuild/issues/575 . 

There are a few upstream problems: 
* https://github.com/oxfordcontrol/osqp-matlab/issues/27
* https://github.com/oxfordcontrol/osqp-matlab/pull/34

But for the time being the repo contains local workaround that should be easy to revert once the issue are solved. 

I already tested the bindings with the example https://github.com/oxfordcontrol/osqp-matlab/blob/master/examples/easy_lasso_qp.m and they seem to be working fine. Once this is merged, I will release the version `v0.6.2.0` .

### CODE REVIEW

**Comment by nunoguedelha** [comment]
Hi @traversaro , I had reviewed most of the changes in this PR, so I thought I could finish the review even if we close https://github.com/robotology/robotology-superbuild/issues/575 and this PR without merging it. What do you think?

**Comment by traversaro** [comment]
Ok!

**Comment by traversaro** [comment]
Given that there are some uses of osqp-matlab in the lab outside of https://github.com/dic-iit/matlab-whole-body-simulator, I think we can proceed with this PRs.

File: README.md [hunk 1]
```diff
@@ -1 +1,53 @@
...
+## Usage
+
+How to use this repo
+
+1. Install osqp in your system, and make sure that it can be find by CMake.
+
+2. Clone the repository
+
+~~~
+git clone git://github.com/dic-iit/osqp-matlab-cmake-buildsystem
```
Comment (nunoguedelha) [inline README.md:23]: Shouldn't this be `git@github.com:dic-iit/osqp-matlab-cmake-buildsystem.git` or `https://github.com/dic-iit/osqp-matlab-cmake-buildsystem`? seems a mix of both :)

File: 3rdparty/ycm-0.12/FindOctave.cmake [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+
+  execute_process ( COMMAND ${OCTAVE_CONFIG_EXECUTABLE} -p OCTLIBDIR
+                    OUTPUT_VARIABLE OCTAVE_LIBRARIES_PATHS
+                    OUTPUT_STRIP_TRAILING_WHITESPACE )
+
+  execute_process ( COMMAND ${OCTAVE_CONFIG_EXECUTABLE} -p OCTFILEDIR
+                    OUTPUT_VARIABLE OCTAVE_OCT_FILE_DIR
+                    OUTPUT_STRIP_TRAILING_WHITESPACE )
+
+  execute_process ( COMMAND ${OCTAVE_CONFIG_EXECUTABLE} -p OCTLIBDIR
```
[the remark this answers is not in the document]
Comment (traversaro) [reply 3rdparty/ycm-0.12/FindOctave.cmake:73]: All the files in the `3rdparty` directory are vendored from another project (YCM, https://github.com/robotology/ycm/blob/v0.12.1/3rdparty/cmake-wiki/FindOctave.cmake) that itself got their file from https://gitlab.kitware.com/cmake/community/-/wikis/contrib/modules/FindOctave . As such, if there is any doubt regarding those files appropriate issue should be opened in the corresponding repo, as modifying only the copies would be confusing.

File: README.md (hunk 1 above)
Comment (traversaro) [reply README.md:23]: It seems to work fine, no?

File: 3rdparty/ycm-0.12/FindOctave.cmake [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+    string ( REGEX REPLACE "[0-9]+\\.[0-9]+\\.([0-9]+).*" "\\1" OCTAVE_PATCH_VERSION ${OCTAVE_VERSION_STRING} )
+  endif ()
+endif ()
+find_program( OCTAVE_EXECUTABLE
+              HINTS ${OCTAVE_BIN_PATHS}
+              NAMES octave
+            )
+find_library( OCTAVE_OCTINTERP_LIBRARY
+              NAMES octinterp liboctinterp
+              HINTS ${OCTAVE_LIBRARIES_PATHS}
```
[the remark this answers is not in the document]
Comment (traversaro) [reply 3rdparty/ycm-0.12/FindOctave.cmake:93]: See https://github.com/dic-iit/osqp-matlab-cmake-buildsystem/pull/1#discussion_r597140833 .

File: README.md (hunk 1 above)
Comment (traversaro) [reply README.md]: Anyhow, I changed it in https://github.com/dic-iit/osqp-matlab-cmake-buildsystem/pull/1/commits/17b1aca369cbb37cb2e195440553993e5b409d87 to the URL that github suggest when you click on "Clone this repo" .

File: 3rdparty/ycm-0.12/FindOctave.cmake (hunk 2 above)
[the remark this answers is not in the document]
Comment (nunoguedelha) [reply 3rdparty/ycm-0.12/FindOctave.cmake:73]: indeed. In any case if it's a copy, we can say it's out of the scope of this PR.

File: README.md (hunk 1 above)
Comment (nunoguedelha) [reply README.md]: not for me, but it's probably my problem as I realized I don't have a key, I always use https.

**Comment by nunoguedelha** [comment]
The example https://github.com/oxfordcontrol/osqp-matlab/blob/master/examples/easy_lasso_qp.m is also working for me after building the repo.

LGTM.

**Review by nunoguedelha — APPROVED** [review]

**Pull request merged**